# Crédit  

In [3]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns
import warnings 
warnings.filterwarnings('ignore')

In [4]:
import os 
os.chdir("/Users/Bryan/Desktop/Openclasseroom/P8_Joachim_Bryan")

In [5]:
Info_CB = pd.read_csv("application_record.csv")
Info_CB

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
0       5008804           M            Y               Y             0   
1       5008805           M            Y               Y             0   
2       5008806           M            Y               Y             0   
3       5008808           F            N               Y             0   
4       5008809           F            N               Y             0   
...         ...         ...          ...             ...           ...   
438552  6840104           M            N               Y             0   
438553  6840222           F            N               N             0   
438554  6841878           F            N               N             0   
438555  6842765           F            N               Y             0   
438556  6842885           F            N               Y             0   

        AMT_INCOME_TOTAL      NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
0               427500.0               Working               Higher education   
1               427500.0               Working               Higher education   
2               112500.0               Working  Secondary / secondary special   
3               270000.0  Commercial associate  Secondary / secondary special   
4               270000.0  Commercial associate  Secondary / secondary special   
...                  ...                   ...                            ...   
438552          135000.0             Pensioner  Secondary / secondary special   
438553          103500.0               Working  Secondary / secondary special   
438554           54000.0  Commercial associate               Higher education   
438555           72000.0             Pensioner  Secondary / secondary special   
438556          121500.0               Working  Secondary / secondary special   

          NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
0             Civil marriage   Rented apartment      -12005          -4542   
1             Civil marriage   Rented apartment      -12005          -4542   
2                    Married  House / apartment      -21474          -1134   
3       Single / not married  House / apartment      -19110          -3051   
4       Single / not married  House / apartment      -19110          -3051   
...                      ...                ...         ...            ...   
438552             Separated  House / apartment      -22717         365243   
438553  Single / not married  House / apartment      -15939          -3007   
438554  Single / not married       With parents       -8169           -372   
438555               Married  House / apartment      -21673         365243   
438556               Married  House / apartment      -18858          -1201   

        FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  \
0                1                1           0           0             NaN   
1                1                1           0           0             NaN   
2                1                0           0           0  Security staff   
3                1                0           1           1     Sales staff   
4                1                0           1           1     Sales staff   
...            ...              ...         ...         ...             ...   
438552           1                0           0           0             NaN   
438553           1                0           0           0        Laborers   
438554           1                1           0           0     Sales staff   
438555           1                0           0           0             NaN   
438556           1                0           1           0     Sales staff   

        CNT_FAM_MEMBERS  
0                   2.0  
1                   2.0  
2                   2.0  
3                   1.0  
4                   1.0  
...                 ...  
438552              1.0  
438553              1.0  
438554              1.0  
43855

Contexte

Les cartes de crédit sont une méthode de contrôle des risques courante dans le secteur financier. Elle utilise les informations et les données personnelles soumises par les demandeurs de carte de crédit pour prédire la probabilité de futurs défauts de paiement et d'emprunts par carte de crédit. La banque est en mesure de décider si elle émet une carte de crédit au demandeur. Les scores de crédit permettent de quantifier objectivement l'ampleur du risque.
 
En général, les cartes de crédit sont basées sur des données historiques. Lorsqu'elles sont confrontées à de grandes fluctuations économiques. Les modèles antérieurs peuvent perdre leur pouvoir prédictif initial. Le modèle logistique est une méthode courante de notation de crédit. Parce que le modèle logistique convient aux tâches de classification binaire et peut calculer les coefficients de chaque caractéristique. Afin de faciliter la compréhension et l'exploitation, la carte de pointage multipliera le coefficient de régression logistique par une certaine valeur (comme 100) et l'arrondira.
 
Actuellement, avec le développement des algorithmes d'apprentissage automatique. Des méthodes plus prédictives telles que Boosting, Random Forest et Support Vector Machines ont été introduites dans le scoring des cartes de crédit. Cependant, ces méthodes manquent souvent de transparence. Il peut être difficile de fournir aux clients et aux régulateurs la raison du rejet ou de l'acceptation.

Tâche

Construisez un modèle d'apprentissage automatique pour prédire si un candidat est un "bon" ou un "mauvais" client, contrairement aux autres tâches, la définition de "bon" ou "mauvais" n'est pas donnée. Vous devez utiliser une technique, telle que l'analyse du millésime, pour construire votre étiquette. De plus, le déséquilibre des données est un problème important dans cette tâche.

Contenu et explication

Il y a deux tables qui peuvent être fusionnées par ID :

Nom de la fonction Explication Remarques  
ID Numéro de client	
CODE_GENDER Sexe	
FLAG_OWN_CAR Y a-t-il une voiture ?	
FLAG_OWN_REALTY Y a-t-il un bien immobilier ?	
CNT_CHILDREN Nombre d'enfants	
AMT_INCOME_TOTAL Revenu annuel	
NAME_INCOME_TYPE Catégorie de revenu	
NAME_EDUCATION_TYPE Niveau d'éducation	
NAME_FAMILY_STATUS Situation familiale	
NAME_HOUSING_TYPE Mode de vie	
DAYS_BIRTH Anniversaire Compte à rebours à partir du jour actuel (0), -1 signifie hier
DAYS_EMPLOYED Date de début de l'emploi Compte à rebours à partir de la date actuelle (0). S'il est positif, cela signifie que la personne est actuellement au chômage.
FLAG_MOBIL Y a-t-il un téléphone mobile ?	
FLAG_WORK_PHONE Y a-t-il un téléphone professionnel ?	
FLAG_PHONE Y a-t-il un téléphone ?	
FLAG_EMAIL Existe-t-il une adresse électronique ?	
OCCUPATION_TYPE Profession	
CNT_FAM_MEMBERS Taille de la famille	
credit_record.csv		
Nom de la caractéristique Explication Remarques
ID Numéro de client	
MONTHS_BALANCE Mois de l'enregistrement Le mois des données extraites est le point de départ, à l'envers, 0 est le mois en cours, -1 est le mois précédent, etc.
STATUS Statut 0 : arriéré de 1 à 29 jours 1 : arriéré de 30 à 59 jours 2 : arriéré de 60 à 89 jours 3 : arriéré de 90 à 119 jours 4 : arriéré de 120 à 149 jours 5 : arriéré ou créances irrécouvrables, radiations depuis plus de 150 jours C : remboursé ce mois-là X : pas de prêt pour ce mois-là


## Analyse univarié du jeux de donnée 1

### Info globale

In [202]:
Info_CB.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 438557 entries, 0 to 438556
Data columns (total 18 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   ID                   438557 non-null  int64  
 1   CODE_GENDER          438557 non-null  object 
 2   FLAG_OWN_CAR         438557 non-null  object 
 3   FLAG_OWN_REALTY      438557 non-null  object 
 4   CNT_CHILDREN         438557 non-null  int64  
 5   AMT_INCOME_TOTAL     438557 non-null  float64
 6   NAME_INCOME_TYPE     438557 non-null  object 
 7   NAME_EDUCATION_TYPE  438557 non-null  object 
 8   NAME_FAMILY_STATUS   438557 non-null  object 
 9   NAME_HOUSING_TYPE    438557 non-null  object 
 10  DAYS_BIRTH           438557 non-null  int64  
 11  DAYS_EMPLOYED        438557 non-null  int64  
 12  FLAG_MOBIL           438557 non-null  int64  
 13  FLAG_WORK_PHONE      438557 non-null  int64  
 14  FLAG_PHONE           438557 non-null  int64  
 15  FLAG_EMAIL       

### Variable ID 

>Identifiant des clients de l'échantillon  
**438557** valeurs de type int64

In [203]:
print("Nombre de client : ",Info_CB.ID.nunique(),"\n")
print("Nombre d'observation :",Info_CB.shape[0])

Nombre de client :  438510 

Nombre d'observation : 438557


> On remarque que il y a un différentielle de 47 clients en plus.  
Ces clients sont des doublons car les clients sont au nombre de **438510**

In [204]:
# Clients avec le meme ID :

Same_ID = Info_CB.loc[Info_CB.ID.isin(Info_CB.ID.value_counts()[Info_CB.ID.value_counts() == 2].index)
            ,:]

# On stock les index : 
Index_same = Same_ID.sort_values(by = "ID").index

In [205]:
# On change les ID pour les doublons par l'ID max + 1 : on commence a 8000000

for i in list(range(0,93,2)):
    Info_CB.loc[Index_same[i],'ID'] = Info_CB.ID.max()+1

In [206]:
# Nombre de clients : 
print("Nombre de client : ",Info_CB.ID.nunique(),"\n")

Nombre de client :  438557 



> On avait 47 clients qui partageaient leur ID avec d'autres clients.  
On a donc créé les ID des 47 clients a partir de 8000000.  

**Résultat : nous avons bien 438557 clients dans cette échantillon.**

**Coté variables :**
>+ **Same_ID :** Data composée des ID en double(94). ( Après rectification Same_ID à 47 client ) 
+ **Index_same :** Indexe des clients qui ont un doublon d'ID
+ **Info_CB :** Data de travail 

### Variable CODE_GENDER

> Sexe des clients de l'échantillon  
**438557** valeur de type Object (string)

In [207]:
print(Info_CB.CODE_GENDER.unique(),"\n")

print("Pour Masculin : ",Info_CB.CODE_GENDER.unique()[0]," ",
      np.round((Info_CB.CODE_GENDER.value_counts()[1]/Info_CB.shape[0])*100,1)
      ,"%\t",Info_CB.CODE_GENDER.value_counts()[1],"\n")

print("Pour Féminin : ",Info_CB.CODE_GENDER.unique()[1]," ",
      np.round((Info_CB.CODE_GENDER.value_counts()[0]/Info_CB.shape[0])*100,1),"%\t",
      Info_CB.CODE_GENDER.value_counts()[0])

['M' 'F'] 

Pour Masculin :  M   32.9 %	 144117 

Pour Féminin :  F   67.1 %	 294440


### Variable FLAG_OWN_CAR

> 

In [208]:
print(Info_CB.FLAG_OWN_CAR.unique(),"\n")

print("Possède une voiture : ",Info_CB.FLAG_OWN_CAR.unique()[0]," ",
      np.round((Info_CB.FLAG_OWN_CAR.value_counts()[1]/Info_CB.shape[0])*100,1)
      ,"%\t",Info_CB.FLAG_OWN_CAR.value_counts()[1],"\n")

print("Ne possède pas de voiture : ",Info_CB.FLAG_OWN_CAR.unique()[1]," ",
      np.round((Info_CB.FLAG_OWN_CAR.value_counts()[0]/Info_CB.shape[0])*100,1),"%\t",
      Info_CB.FLAG_OWN_CAR.value_counts()[0])

['Y' 'N'] 

Possède une voiture :  Y   37.2 %	 163098 

Ne possède pas de voiture :  N   62.8 %	 275459


### Variable FLAG_OWN_REALTY

> 

In [209]:
print(Info_CB.FLAG_OWN_REALTY.unique(),"\n")

print("Possède un bien immobilier : ",Info_CB.FLAG_OWN_REALTY.unique()[0]," ",
      np.round((Info_CB.FLAG_OWN_REALTY.value_counts()[1]/Info_CB.shape[0])*100,1)
      ,"%\t",Info_CB.FLAG_OWN_REALTY.value_counts()[1],"\n")

print("Ne possède pas de bien immobilier : ",Info_CB.FLAG_OWN_REALTY.unique()[1]," ",
      np.round((Info_CB.FLAG_OWN_REALTY.value_counts()[0]/Info_CB.shape[0])*100,1),"%\t",
      Info_CB.FLAG_OWN_REALTY.value_counts()[0])

['Y' 'N'] 

Possède un bien immobilier :  Y   30.7 %	 134483 

Ne possède pas de bien immobilier :  N   69.3 %	 304074


### Variable CNT_CHILDREN

>

In [210]:
Info_CB.CNT_CHILDREN.unique()

array([ 0,  1,  3,  2,  4,  5, 14, 19,  7,  9, 12,  6])

> On s'aperçoit que le nombre d'enfant varie entre 0 et 19.  
+ 19 enfants parait démesuré quoique pas impossible ( 14,12 également )  

>Regardons pour 19 et 14 

In [216]:
Info_CB[Info_CB.AMT_INCOME_TOTAL == 135000.0]

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
13      5008819           M            Y               Y             0   
14      5008820           M            Y               Y             0   
15      5008821           M            Y               Y             0   
16      5008822           M            Y               Y             0   
17      5008823           M            Y               Y             0   
...         ...         ...          ...             ...           ...   
438548  6839936           M            Y               Y             1   
438549  6840098           F            N               Y             0   
438550  6840100           F            N               Y             0   
438551  6840102           F            N               Y             0   
438552  6840104           M            N               Y             0   

        AMT_INCOME_TOTAL      NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
13              135000.0  Commercial associate  Secondary / secondary special   
14              135000.0  Commercial associate  Secondary / secondary special   
15              135000.0  Commercial associate  Secondary / secondary special   
16              135000.0  Commercial associate  Secondary / secondary special   
17              135000.0  Commercial associate  Secondary / secondary special   
...                  ...                   ...                            ...   
438548          135000.0               Working  Secondary / secondary special   
438549          135000.0             Pensioner  Secondary / secondary special   
438550          135000.0             Pensioner  Secondary / secondary special   
438551          135000.0             Pensioner  Secondary / secondary special   
438552          135000.0             Pensioner  Secondary / secondary special   

       NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
13                Married  House / apartment      -17778          -1194   
14                Married  House / apartment      -17778          -1194   
15                Married  House / apartment      -17778          -1194   
16                Married  House / apartment      -17778          -1194   
17                Married  House / apartment      -17778          -1194   
...                   ...                ...         ...            ...   
438548            Married  House / apartment      -12569          -2095   
438549          Separated  House / apartment      -22717         365243   
438550          Separated  House / apartment      -22717         365243   
438551          Separated  House / apartment      -22717         365243   
438552          Separated  House / apartment      -22717         365243   

        FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  \
13               1                0           0           0        Laborers   
14               1                0           0           0        Laborers   
15               1                0           0           0        Laborers   
16               1                0           0           0        Laborers   
17               1                0           0           0        Laborers   
...            ...              ...         ...         ...             ...   
438548           1                0           0           0        Laborers   
438549           1                0           0           0             NaN   
438550           1                0           0           0             NaN   
438551           1                0           0           0             NaN   
438552           1                0           0           0             NaN   

        CNT_FAM_MEMBERS  
13                  2.0  
14                  2.0  
15                  2.0  
16                  2.0  
17                  2.0  
...                 ...  
438548              3.0  
438549              1.0  
438550              1.0  
438551              1.0  
438552         

In [217]:
Info_CB[Info_CB.CNT_CHILDREN.isin([19,14])]

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
20441  5061207           M            Y               Y            14   
20442  5061210           M            Y               Y            14   
20443  5061211           M            Y               Y            14   
36388  5105054           F            N               Y            19   

       AMT_INCOME_TOTAL NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
20441          225000.0          Working  Secondary / secondary special   
20442          225000.0          Working  Secondary / secondary special   
20443          225000.0          Working  Secondary / secondary special   
36388          112500.0          Working  Secondary / secondary special   

         NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
20441             Separated  House / apartment      -17754          -1689   
20442             Separated  House / apartment      -17754          -1689   
20443             Separated  House / apartment      -17754          -1689   
36388  Single / not married  House / apartment      -11057          -1853   

       FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL  \
20441           1                0           0           0   
20442           1                0           0           0   
20443           1                0           0           0   
36388           1                1           1           0   

            OCCUPATION_TYPE  CNT_FAM_MEMBERS  
20441               Drivers             15.0  
20442               Drivers             15.0  
20443               Drivers             15.0  
36388  Waiters/barmen staff             20.0

> On Observe des doublons sur les caractéristiques, avec des identifiants différents.  

>**On va transformer la data en mettant les ID en index puis en faisant un drop_duplicates().**

In [125]:
# On supprime les observations avec un ID different mais aux information identiques.
Info_CB = Info_CB.set_index('ID').drop_duplicates().reset_index()
Info_CB.shape

(90085, 18)

>+ On passe de **438 557** à **90 085** clients.  
+ Soit une perte de 348 472 observations...  
+ Soit prés de 80% des données.  

In [167]:
Info_CB.CNT_FAM_MEMBERS.value_counts()

2.0     47397
1.0     18389
3.0     15631
4.0      7483
5.0      1047
6.0       106
7.0        24
9.0         2
8.0         2
11.0        1
15.0        1
14.0        1
20.0        1
Name: CNT_FAM_MEMBERS, dtype: int64

### Variable AMT_INCOME_TOTAL

> Revenu annuelle du client

In [138]:
Info_CB.AMT_INCOME_TOTAL.describe().astype(int)

count      90085
mean      178700
std       102462
min        26100
25%       112500
50%       157500
75%       225000
max      6750000
Name: AMT_INCOME_TOTAL, dtype: int64

### Variable NAME_INCOME_TYPE

In [147]:
print(Info_CB.NAME_INCOME_TYPE.describe(),"\n")
print(Info_CB.NAME_INCOME_TYPE.value_counts())

count       90085
unique          5
top       Working
freq        46633
Name: NAME_INCOME_TYPE, dtype: object 

Working                 46633
Commercial associate    20957
Pensioner               15839
State servant            6650
Student                     6
Name: NAME_INCOME_TYPE, dtype: int64


### Variable NAME_EDUCATION_TYPE

In [148]:
print(Info_CB.NAME_EDUCATION_TYPE.describe(),"\n")
print(Info_CB.NAME_EDUCATION_TYPE.value_counts())

count                             90085
unique                                5
top       Secondary / secondary special
freq                              63479
Name: NAME_EDUCATION_TYPE, dtype: object 

Secondary / secondary special    63479
Higher education                 22477
Incomplete higher                 3100
Lower secondary                    971
Academic degree                     58
Name: NAME_EDUCATION_TYPE, dtype: int64


### Variable NAME_FAMILY_STATUS

In [149]:
print(Info_CB.NAME_FAMILY_STATUS.describe(),"\n")
print(Info_CB.NAME_FAMILY_STATUS.value_counts())

count       90085
unique          5
top       Married
freq        59825
Name: NAME_FAMILY_STATUS, dtype: object 

Married                 59825
Single / not married    12708
Civil marriage           8009
Separated                5492
Widow                    4051
Name: NAME_FAMILY_STATUS, dtype: int64


### Variable NAME_HOUSING_TYPE

In [151]:
print(Info_CB.NAME_HOUSING_TYPE.describe(),"\n")
print(Info_CB.NAME_HOUSING_TYPE.value_counts())

count                 90085
unique                    6
top       House / apartment
freq                  80335
Name: NAME_HOUSING_TYPE, dtype: object 

House / apartment      80335
With parents            4156
Municipal apartment     3130
Rented apartment        1373
Office apartment         790
Co-op apartment          301
Name: NAME_HOUSING_TYPE, dtype: int64


### Variable OCCUPATION_TYPE

In [153]:
print(Info_CB.OCCUPATION_TYPE.describe(),"\n")
print(Info_CB.OCCUPATION_TYPE.value_counts())

count        62608
unique          18
top       Laborers
freq         16624
Name: OCCUPATION_TYPE, dtype: object 

Laborers                 16624
Sales staff               9123
Core staff                8091
Managers                  6675
Drivers                   5860
High skill tech staff     3357
Accountants               3034
Medicine staff            2484
Cooking staff             1764
Security staff            1740
Cleaning staff            1222
Private service staff      750
Low-skill Laborers         569
Secretaries                412
Waiters/barmen staff       393
Realty agents              218
HR staff                   167
IT staff                   125
Name: OCCUPATION_TYPE, dtype: int64


### Reste des variables numérique 
> Il reste :
+ DAYS_BIRTH           
+ DAYS_EMPLOYED        
+ FLAG_MOBIL          
+ FLAG_WORK_PHONE      
+ FLAG_PHONE          
+ FLAG_EMAIL  
+ CNT_FAM_MEMBERS

In [157]:
Info_CB.describe().astype(int)

ID  CNT_CHILDREN  AMT_INCOME_TOTAL  DAYS_BIRTH  DAYS_EMPLOYED  \
count    90085         90085             90085       90085          90085   
mean   5907759             0            178700      -15901          61589   
std     517295             0            102462        4255         139450   
min    5008804             0             26100      -25201         -17531   
25%    5465838             0            112500      -19429          -2870   
50%    5952121             0            157500      -15566          -1299   
75%    6289034             1            225000      -12339           -324   
max    7995770            19           6750000       -7489         365243   

       FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL  CNT_FAM_MEMBERS  
count       90085            90085       90085       90085            90085  
mean            1                0           0           0                2  
std             0                0           0           0                0  
min             1                0           0           0                1  
25%             1                0           0           0                2  
50%             1                0           0           0                2  
75%             1                0           1           0                3  
max             1                1           1           1               20

In [163]:
Info_CB.FLAG_EMAIL.value_counts()

0    81056
1     9029
Name: FLAG_EMAIL, dtype: int64

In [164]:
Info_CB.FLAG_PHONE.value_counts()

0    65357
1    24728
Name: FLAG_PHONE, dtype: int64

In [165]:
Info_CB.FLAG_WORK_PHONE.value_counts()

0    71802
1    18283
Name: FLAG_WORK_PHONE, dtype: int64

In [168]:
Info_CB.FLAG_MOBIL.value_counts()

1    90085
Name: FLAG_MOBIL, dtype: int64

## Analyse univarié du jeu de donnée 2

In [169]:
Dossier_Credit = pd.read_csv("credit_record.csv")
Dossier_Credit

ID  MONTHS_BALANCE STATUS
0        5001711               0      X
1        5001711              -1      0
2        5001711              -2      0
3        5001711              -3      0
4        5001712               0      C
...          ...             ...    ...
1048570  5150487             -25      C
1048571  5150487             -26      C
1048572  5150487             -27      C
1048573  5150487             -28      C
1048574  5150487             -29      C

[1048575 rows x 3 columns]

In [170]:
Dossier_Credit.STATUS.value_counts()

C    442031
0    383120
X    209230
1     11090
5      1693
2       868
3       320
4       223
Name: STATUS, dtype: int64

In [172]:
Dossier_Credit.MONTHS_BALANCE.unique()

array([  0,  -1,  -2,  -3,  -4,  -5,  -6,  -7,  -8,  -9, -10, -11, -12,
       -13, -14, -15, -16, -17, -18, -19, -20, -21, -22, -23, -24, -25,
       -26, -27, -28, -29, -30, -31, -32, -33, -34, -35, -36, -37, -38,
       -39, -40, -41, -42, -43, -44, -45, -46, -47, -48, -49, -50, -51,
       -52, -53, -54, -55, -56, -57, -58, -59, -60])

## Merge des deux jeu de données 

In [174]:
Info = Info_CB.merge(Dossier_Credit,"left", on = "ID")
Info.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 299549 entries, 0 to 299548
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   ID                   299549 non-null  int64  
 1   CODE_GENDER          299549 non-null  object 
 2   FLAG_OWN_CAR         299549 non-null  object 
 3   FLAG_OWN_REALTY      299549 non-null  object 
 4   CNT_CHILDREN         299549 non-null  int64  
 5   AMT_INCOME_TOTAL     299549 non-null  float64
 6   NAME_INCOME_TYPE     299549 non-null  object 
 7   NAME_EDUCATION_TYPE  299549 non-null  object 
 8   NAME_FAMILY_STATUS   299549 non-null  object 
 9   NAME_HOUSING_TYPE    299549 non-null  object 
 10  DAYS_BIRTH           299549 non-null  int64  
 11  DAYS_EMPLOYED        299549 non-null  int64  
 12  FLAG_MOBIL           299549 non-null  int64  
 13  FLAG_WORK_PHONE      299549 non-null  int64  
 14  FLAG_PHONE           299549 non-null  int64  
 15  FLAG_EMAIL       

In [175]:
Info.OCCUPATION_TYPE.fillna(0,inplace = True)

In [179]:
Info_sans_nan = Info.dropna()
Info_sans_nan.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 219173 entries, 0 to 298402
Data columns (total 20 columns):
 #   Column               Non-Null Count   Dtype  
---  ------               --------------   -----  
 0   ID                   219173 non-null  int64  
 1   CODE_GENDER          219173 non-null  object 
 2   FLAG_OWN_CAR         219173 non-null  object 
 3   FLAG_OWN_REALTY      219173 non-null  object 
 4   CNT_CHILDREN         219173 non-null  int64  
 5   AMT_INCOME_TOTAL     219173 non-null  float64
 6   NAME_INCOME_TYPE     219173 non-null  object 
 7   NAME_EDUCATION_TYPE  219173 non-null  object 
 8   NAME_FAMILY_STATUS   219173 non-null  object 
 9   NAME_HOUSING_TYPE    219173 non-null  object 
 10  DAYS_BIRTH           219173 non-null  int64  
 11  DAYS_EMPLOYED        219173 non-null  int64  
 12  FLAG_MOBIL           219173 non-null  int64  
 13  FLAG_WORK_PHONE      219173 non-null  int64  
 14  FLAG_PHONE           219173 non-null  int64  
 15  FLAG_EMAIL       

## Enregistrement de la Data de travail

In [186]:
Info_sans_nan.to_csv(r'Info.csv',index=False,header = True)
Info_CB.to_csv(r'Info_client.csv',index = False, header = True)
Info.to_csv(r'Info_avec_nan.csv', index = False, header = True)

In [15]:
Info_CB["NAME_HOUSING_TYPE"].unique()

ID CODE_GENDER FLAG_OWN_CAR FLAG_OWN_REALTY  CNT_CHILDREN  \
0       5008804           M            Y               Y             0   
1       5008805           M            Y               Y             0   
2       5008806           M            Y               Y             0   
3       5008808           F            N               Y             0   
4       5008809           F            N               Y             0   
...         ...         ...          ...             ...           ...   
438552  6840104           M            N               Y             0   
438553  6840222           F            N               N             0   
438554  6841878           F            N               N             0   
438555  6842765           F            N               Y             0   
438556  6842885           F            N               Y             0   

        AMT_INCOME_TOTAL      NAME_INCOME_TYPE            NAME_EDUCATION_TYPE  \
0               427500.0               Working               Higher education   
1               427500.0               Working               Higher education   
2               112500.0               Working  Secondary / secondary special   
3               270000.0  Commercial associate  Secondary / secondary special   
4               270000.0  Commercial associate  Secondary / secondary special   
...                  ...                   ...                            ...   
438552          135000.0             Pensioner  Secondary / secondary special   
438553          103500.0               Working  Secondary / secondary special   
438554           54000.0  Commercial associate               Higher education   
438555           72000.0             Pensioner  Secondary / secondary special   
438556          121500.0               Working  Secondary / secondary special   

          NAME_FAMILY_STATUS  NAME_HOUSING_TYPE  DAYS_BIRTH  DAYS_EMPLOYED  \
0             Civil marriage   Rented apartment      -12005          -4542   
1             Civil marriage   Rented apartment      -12005          -4542   
2                    Married  House / apartment      -21474          -1134   
3       Single / not married  House / apartment      -19110          -3051   
4       Single / not married  House / apartment      -19110          -3051   
...                      ...                ...         ...            ...   
438552             Separated  House / apartment      -22717         365243   
438553  Single / not married  House / apartment      -15939          -3007   
438554  Single / not married       With parents       -8169           -372   
438555               Married  House / apartment      -21673         365243   
438556               Married  House / apartment      -18858          -1201   

        FLAG_MOBIL  FLAG_WORK_PHONE  FLAG_PHONE  FLAG_EMAIL OCCUPATION_TYPE  \
0                1                1           0           0             NaN   
1                1                1           0           0             NaN   
2                1                0           0           0  Security staff   
3                1                0           1           1     Sales staff   
4                1                0           1           1     Sales staff   
...            ...              ...         ...         ...             ...   
438552           1                0           0           0             NaN   
438553           1                0           0           0        Laborers   
438554           1                1           0           0     Sales staff   
438555           1                0           0           0             NaN   
438556           1                0           1           0     Sales staff   

        CNT_FAM_MEMBERS  
0                   2.0  
1                   2.0  
2                   2.0  
3                   1.0  
4                   1.0  
...                 ...  
438552              1.0  
438553              1.0  
438554              1.0  
43855

In [18]:
Info_CB["OCCUPATION_TYPE"].unique()

array([nan, 'Security staff', 'Sales staff', 'Accountants', 'Laborers',
       'Managers', 'Drivers', 'Core staff', 'High skill tech staff',
       'Cleaning staff', 'Private service staff', 'Cooking staff',
       'Low-skill Laborers', 'Medicine staff', 'Secretaries',
       'Waiters/barmen staff', 'HR staff', 'Realty agents', 'IT staff'],
      dtype=object)